src/ir/effects.h

```diff
@@ -35,6 +35,21 @@ struct EffectAnalyzer
     }
   }
 
+  enum SideEffects : uint32_t {
```

Comment:
perhaps this and the code that uses it below could be at the bottom of this class, and with a comment `// C API` before them?